### PR TITLE
add status.Phase check for GetStorageClassClaimConfig

### PR DESCRIPTION
**Pull request by Madhu-1**
check the status.Phase for GetStorageClassClaimConfig based on the Phase return success or failure response.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

/assign @umangachapagain 
/assign @nb-ohad

### CODE REVIEW

File: services/provider/server/server.go [hunk 1]
```diff
@@ -530,103 +530,112 @@ func (s *OCSProviderServer) GetStorageClassClaimConfig(ctx context.Context, req
...
+	// Verify Status.Phase
+	msg := fmt.Sprintf("storage class claim %q for %q is in %q phase", req.StorageClassClaimName, req.StorageConsumerUUID, storageClassClaim.Status.Phase)
+	switch storageClassClaim.Status.Phase {
+	case ocsv1alpha1.StorageClassClaimFailed:
+		return nil, status.Error(codes.Internal, msg)
+	case ocsv1alpha1.StorageClassClaimInitializing:
+		return nil, status.Error(codes.Unavailable, msg)
+	case ocsv1alpha1.StorageClassClaimCreating:
+		return nil, status.Error(codes.Unavailable, msg)
+	case ocsv1alpha1.StorageClassClaimReady:
```
Comment (nb-ohad) [inline services/provider/server/server.go:542]: This nesting indent the entirety of the code in this function and produce a switch case inside a switch case. 

```suggestion
        if storageClassClaim.Status.Phase != ocsv1alpha1.StorageClassClaimReady {
		switch storageClassClaim.Status.Phase {
		case ocsv1alpha1.StorageClassClaimInitializing:
			return nil, status.Error(codes.Unavailable, msg)
		case ocsv1alpha1.StorageClassClaimCreating:
			return nil, status.Error(codes.Unavailable, msg)
		default:
			return nil, status.Error(codes.Internal, msg)
	}
	
	// All of the main code of the function goes here without additional indentation 
```

**Review by nb-ohad — CHANGES_REQUESTED** [review]

File: services/provider/server/server.go (hunk 1 above)
Comment (Madhu-1) [reply services/provider/server/server.go]: sounds better :+1:

**Pull request merged**